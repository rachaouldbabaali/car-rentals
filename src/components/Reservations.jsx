import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchReservations } from '../redux/reservations/reservationSlice';

function Reservations() {
  const { reservations } = useSelector((store) => store.reservations);

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchReservations());
  }, [dispatch]);

  return (
    <section className="bg-[#95BF02] fixed top-0 bottom-0 left-0 right-0">
      <h1 className="text-center">Reservations</h1>

      <table className="mx-auto table-auto border-collapse border border-slate-500">
        <thead>
          <tr>
            <th className="border border-slate-600">Start Date</th>
            <th className="border border-slate-600">End Date</th>
            <th className="border border-slate-600">City</th>
            <th className="border border-slate-600">Vehicles</th>
          </tr>
        </thead>
        <tbody>
          {reservations && reservations.map((reservation) => (
            <tr key={reservation.id}>
              <td className="border border-slate-600">{reservation.rentDate}</td>
              <td className="border border-slate-600">{reservation.returnDate}</td>
              <td className="border border-slate-600">{reservation.city}</td>
              <td className="border border-slate-600">{reservation.car.name}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </section>
  );
}

export default Reservations;
