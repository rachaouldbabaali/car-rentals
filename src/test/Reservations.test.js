import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import renderer from 'react-test-renderer';
import myReservations from './reservationsData';
import Reservations from '../components/Reservations';

jest.mock('axios');

const reducer = (state = { reservations: { reservations: myReservations } }) => state;

const store = configureStore({ reducer });

describe('Should render correctly', () => {
  it('It should render Reservations', async () => {
    axios.get.mockResolvedValue({ data: myReservations });
    const tree = renderer
      .create(
        <Provider store={store}>
          <Reservations />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
