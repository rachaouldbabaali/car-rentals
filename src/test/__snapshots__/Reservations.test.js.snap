// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly It should get all mission loaded 1`] = `
<section
  className=" bg-[#95BF02] fixed top-0 overflow-auto left-0 h-full w-full p-2"
>
  <h1
    className="text-center"
  >
    Reservations
  </h1>
  <p
    className="text-center"
  />
  <table
    className="mx-auto table-auto border-collapse border border-slate-500"
  >
    <thead>
      <tr>
        <th
          className="border border-slate-600"
        >
          Start Date
        </th>
        <th
          className="border border-slate-600"
        >
          End Date
        </th>
        <th
          className="border border-slate-600"
        >
          City
        </th>
        <th
          className="border border-slate-600"
        >
          Vehicles
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Dar es Salaam
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2023-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2023-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Algeria
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2021-03-07
        </td>
        <td
          className="border border-slate-600"
        >
          Kigali
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          1900-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          1980-03-23
        </td>
        <td
          className="border border-slate-600"
        >
          Africa
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Tanzania
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          World
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Dar es Salaam
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Should render correctly It should get all reservations loaded 1`] = `
<section
  className=" bg-[#95BF02] fixed top-0 overflow-auto left-0 h-full w-full p-2"
>
  <h1
    className="text-center"
  >
    Reservations
  </h1>
  <p
    className="text-center"
  />
  <table
    className="mx-auto table-auto border-collapse border border-slate-500"
  >
    <thead>
      <tr>
        <th
          className="border border-slate-600"
        >
          Start Date
        </th>
        <th
          className="border border-slate-600"
        >
          End Date
        </th>
        <th
          className="border border-slate-600"
        >
          City
        </th>
        <th
          className="border border-slate-600"
        >
          Vehicles
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Dar es Salaam
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2023-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2023-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Algeria
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2021-03-07
        </td>
        <td
          className="border border-slate-600"
        >
          Kigali
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          1900-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          1980-03-23
        </td>
        <td
          className="border border-slate-600"
        >
          Africa
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Tanzania
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          World
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
      <tr>
        <td
          className="border border-slate-600"
        >
          2020-01-21
        </td>
        <td
          className="border border-slate-600"
        >
          2020-04-23
        </td>
        <td
          className="border border-slate-600"
        >
          Dar es Salaam
        </td>
        <td
          className="border border-slate-600"
        />
      </tr>
    </tbody>
  </table>
</section>
`;
